src/sql-parser/tests/testdata/ddl

```diff
@@ -1338,6 +1338,27 @@ CREATE CLUSTER REPLICA default.replica SIZE 'small'
 =>
 CreateClusterReplica(CreateClusterReplicaStatement { of_cluster: Ident("default"), definition: ReplicaDefinition { name: Ident("replica"), options: [Size(Value(String("small")))] } })
 
+parse-statement
+CREATE CLUSTER REPLICA default.replica SIZE 'small', AVAILABILITY ZONE 'a'
+----
+CREATE CLUSTER REPLICA default.replica SIZE 'small', AVAILABILITY ZONE 'a'
+=>
+CreateClusterReplica(CreateClusterReplicaStatement { of_cluster: Ident("default"), definition: ReplicaDefinition { name: Ident("replica"), options: [Size(Value(String("small"))), AvailabilityZone(Value(String("a")))] } })
+
+parse-statement
+CREATE CLUSTER REPLICA default.replica AVAILABILITY ZONE 'a', AVAILABILITY ZONE 'b'
+----
+CREATE CLUSTER REPLICA default.replica AVAILABILITY ZONE 'a', AVAILABILITY ZONE 'b'
+=>
+CreateClusterReplica(CreateClusterReplicaStatement { of_cluster: Ident("default"), definition: ReplicaDefinition { name: Ident("replica"), options: [AvailabilityZone(Value(String("a"))), AvailabilityZone(Value(String("b")))] } })
+
+parse-statement
+CREATE CLUSTER REPLICA default.replica REMOTE ('host1'), AVAILABILITY ZONE 'a'
+----
+CREATE CLUSTER REPLICA default.replica REMOTE ('host1'), AVAILABILITY ZONE 'a'
+=>
+CreateClusterReplica(CreateClusterReplicaStatement { of_cluster: Ident("default"), definition: ReplicaDefinition { name: Ident("replica"), options: [Remote { hosts: [Value(String("host1"))] }, AvailabilityZone(Value(String("a")))] } })
+
```

Comment:
I noticed that this syntax doesn't match the [design doc] anymore. Should we update it?

[design doc]: https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20220413_cluster_replica.md#sql-syntax-and-semantics

Reply:
Nikhil has a PR for that, fortunately #12213 -- I haven't merged it yet but will do so right after this.

Reply:
Hadn't seen that! The main inconsistency I noticed is that the current syntax is `CREATE CLUSTER REPLICA default.replica ...`, whereas the design doc says it should be `CREATE CLUSTER REPLICA replica FOR default ...`. That part is not touched by that PR.

Reply:
I'll fix that, thanks!

Reply:
Updated!